pkg/controllers/disruption/helpers.go

```diff
@@ -119,7 +119,18 @@ func simulateScheduling(ctx context.Context, kubeClient client.Client, cluster *
 		// to proceed disrupting if our scheduling decision relies on nodes that haven't entered a terminal state.
 		if !n.Initialized() {
 			for _, p := range n.Pods {
-				results.PodErrors[p] = fmt.Errorf("would schedule against a non-initialized node %s", n.Name())
+				// Only add a pod scheduling error if it isn't on an already deleting node.
+				// If the pod is on a deleting node, we assume one of two things has already happened:
+				// 1. The node was manually terminated, at which the provisioning controller has scheduled or is scheduling a node
+				//    for the pod.
+				// 2. The node was chosen for a previous disruption command, we assume that the uninitialized node will come up
+				//    for this command, and we assume it will be successful. If it is not successful, the node will become
+				//    not terminating, and we will no longer need to consider these pods.
+				if _, ok := lo.Find(deletingNodePods, func(deleting *v1.Pod) bool {
```

Comment:
How slow is this if you have a lot of deleting pods? Is it worth making this check faster, by just quickly storing the keys once and then just checking if the keys exists or not?